src/vm/analysis/type_checker/natives/conversions.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2013-2020 Blockstack PBC, a public benefit corporation
+// Copyright (C) 2020 Stacks Open Internet Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use vm::representations::SymbolicExpression;
+use vm::types::SequenceSubtype;
+use vm::types::TypeSignature;
+
+use vm::analysis::type_checker::{
+    check_argument_count, CheckErrors, TypeChecker, TypeResult, TypingContext,
+};
+use vm::costs::cost_functions::ClarityCostFunction;
+use vm::costs::runtime_cost;
+
+pub fn check_special_buff_to_int(
+    checker: &mut TypeChecker,
+    args: &[SymbolicExpression],
+    context: &TypingContext,
+) -> TypeResult {
+    check_argument_count(1, args)?;
+    runtime_cost(ClarityCostFunction::BuffToInt, checker, 0)?;
+
+    let collection_type = checker.type_check(&args[0], context)?;
+    match collection_type {
```

Comment:
For these functions, the type checker should do an admission check, rather than an exact type check.

e.g.,

`(buff-to-uint 0x00)` should work, because we'll do left-padding, but the type of that input is `(buff 1)`, which would fail the exact type check.

Instead, you can do something like:

`checker.type_check_expects(&args[0], context, &TypeSignature::SequenceType(SequenceSubType::BufferType(16)))`

Rather than the current match check.

The same applies to `buff-to-uint`.

Reply:
I reverted this file. Checking should now be done in src/vm/analysis/type_checker/natives/mod.rs

Is there a way to test the type checking?

Reply:
Yep -- look at the unit tests in `vm/analysis/type_checker/tests/mod.rs` -- you should be able to add some tests for different input types pretty easily, and test them against either expected errors or expected result types.

Reply:
I added some tests in `vm/analysis/type_checker/tests/mod.rs`